odoo/addons/base/models/ir_attachment.py

```diff
@@ -440,19 +440,7 @@ def check(self, mode, values=None):
         if not (self.env.is_admin() or self.env.user._is_internal()):
             raise AccessError(_("Sorry, you are not allowed to access this document."))
         # collect the records to check (by model)
-        model_ids = defaultdict(set)            # {model_name: set(ids)}
-        if self:
-            # DLE P173: `test_01_portal_attachment`
-            self.env['ir.attachment'].flush_model(['res_model', 'res_id', 'create_uid', 'public', 'res_field'])
-            self._cr.execute('SELECT res_model, res_id, create_uid, public, res_field FROM ir_attachment WHERE id IN %s', [tuple(self.ids)])
-            for res_model, res_id, create_uid, public, res_field in self._cr.fetchall():
-                if public and mode == 'read':
-                    continue
-                if not self.env.is_system() and (res_field or (not res_id and create_uid != self.env.uid)):
-                    raise AccessError(_("Sorry, you are not allowed to access this document."))
```

Comment:
why do you remove these checks, it's there for security reasons

Reply:
Hello @mart-e 

We are not removing them, simply moving them to a dedicated method allowing to hook ourselves easier.

Currently, the rule that says "for an ir.attachment with res_id 0, you have to be the creator to access" is a bit too limiting for a feature we are trying to implement in knowledge.
Which is basically a shared "image bank" for covers, meaning you could use the same attachment for multiple articles (hence the usage of res_id = 0).

So we want to be able to say "for knowledge.article and the cover_image_id field, having read.access to the knowledge.article model allows accessing the attachment with res_id 0".

You can check my review comment here above for the full picture if you are interested.

Thanks!